androidbrowserhelper/src/main/java/com/google/androidbrowserhelper/trusted/TwaLauncher.java

```diff
@@ -222,11 +223,15 @@ private void launchWhenSplashScreenReady(TrustedWebActivityIntentBuilder builder
             return;  // Destroyed while preparing the splash screen (e.g. user closed the app).
         }
         Log.d(TAG, "Launching Trusted Web Activity.");
-        Intent intent = builder.build(mSession);
+        Intent intent = builder.build(mSession).getIntent();
         ContextCompat.startActivity(mContext, intent, null);
+
         // Remember who we connect to as the package that is allowed to delegate notifications
         // to us.
-        TrustedWebActivityService.setVerifiedProvider(mContext, mProviderPackage);
+        Token token = Token.create(mProviderPackage, mContext.getPackageManager());
+        TokenStore tokenStore = new SharedPreferencesTokenStore(mContext);
+        tokenStore.store(token);
```

Comment:
Could you move these three lines into a static method on TokenStore (so that this file only knows about TokenStore, not Token), like `setVerifiedProvider`?

Reply:
Or if we want to make testing/customizing easier:

1. Create an instance method `setVerifiedProvider` on TokenStore.
2. Pass a TokenStore in to TwaLauncher in the constructor.

(though I'm happy to leave this until we actually need to test things).

Reply:
I'm creating a `setVerifiedProvider` instance method. The bit that is a not as nice is that I have to pass a `SharedPreferencesTokenStore` in the constructor, as the interface doesn't have `setVerifiedProvider`